src/builtin.c

```diff
@@ -1186,9 +1186,20 @@ static jv f_string_indexes(jq_state *jq, jv a, jv b) {
 }
 
 static jv f_string_implode(jq_state *jq, jv a) {
+  int i;
+  int len;
   if (jv_get_kind(a) != JV_KIND_ARRAY) {
     return ret_error(a, jv_string("implode input must be an array"));
   }
+  len = jv_array_length(jv_copy(a));
+  for (i = 0; i < len; i++) {
+    jv n = jv_array_get(jv_copy(a), i);
+    if (jv_get_kind(n) != JV_KIND_NUMBER) {
```

Comment:
I think it would also make sense to check `jvp_number_is_nan(n)`, since `(int)nan` yields different results with different compilers/architectures.
Also maybe it could also check `!jv_is_integer(n)`, though we could also just let non-ints round down.

Reply:
Good points. Don’t know about non-ints either, i think in most other such situations jq rounds down? Like indexing? will look into in a few days, out travelibg

Reply:
Fixed. Now error on nan and i added a test for rounding down of non-ints